tests/draft2020-12/id.json

```diff
@@ -204,5 +204,62 @@
                 "valid": true
             }
         ]
+    },
+    {
+        "description": "$id inside an enum is not a real identifier",
+        "comment": "the implementation must not be confused by an $id buried in the enum",
+        "schema": {
+            "$defs": {
+                "id_in_enum": {
+                    "enum": [
+                        {
+                          "$id": "https://localhost:1234/my_identifier.json",
+                          "type": "null"
+                        }
+                    ]
+                },
+                "real_id_in_schema": {
+                    "$id": "https://localhost:1234/my_identifier.json",
+                    "type": "string"
+                },
+                "zzz_id_in_const": {
+                    "const": {
+                        "$id": "https://localhost:1234/my_identifier.json",
+                        "type": "null"
+                    }
+                }
+            },
+            "anyOf": [
+                { "$ref": "#/$defs/id_in_enum" },
+                { "$ref": "https://localhost:1234/my_identifier.json" }
+            ]
+        },
+        "tests": [
+            {
+                "description": "exact match to enum, and type matches",
+                "data": {
+                    "$id": "https://localhost:1234/my_identifier.json",
+                    "type": "null"
+                },
+                "valid": true
+            },
+            {
+                "description": "in implementations that strip $id, this may match either of the $defs",
+                "data": {
+                    "type": "null"
+                },
+                "valid": false
+            },
```

Comment:
You can't just strip out `$id`s like you can with `$anchor`s, so I don't see this catching bugs the same way it would with `$anchor`. What I do is replace the entire schema with an object that represents a reference and is not expressible in JSON. Because it's not expressible in JSON, we can't create a test that would catch that. I think we can drop this test for the "id.json" tests and just keep them for "anchor.json" tests.

```suggestion
```

Reply:
okay, I removed those tests for id/$id.